### DISCUSSION TITLE
WebLaf integration in Netbeans

### DISCUSSION THREAD

**Discussion by albilu** [apache/netbeans#6229] [General]
### Description

I just found this beautiful LAF https://github.com/mgarin/weblaf wich is the second most popular after FlatLaf on github.

Would be very cool to have it on Netbeans. Integration should be similar to what was done for FlatLaf

![image](https://github.com/apache/netbeans/assets/35330562/b9dcf92a-1fc6-4dde-8dbc-ae526ce04369)
![image](https://github.com/apache/netbeans/assets/35330562/91cd86cf-9293-4bfe-803c-94b6c7d0905c)

### Use case/motivation

Increase Netbeans Themes options and adoption.

### Are you willing to submit a pull request?

No

**Comment by neilcsmith-net** [comment]
Someone might want to add as a plugin, but it's GPL, so impossible to add to the Apache NetBeans project.

**Reply by mbien** [reply]
the LAF topic made me look up if Napkin LAF still exists: https://napkinlaf.sourceforge.net/
even the license is compatible :)

**Reply by Chris2011** [reply]
![image](https://github.com/apache/netbeans/assets/795658/65928df9-4789-471e-be47-3bd8fd30e8db)

**Comment by albilu** [comment]
@neilcsmith-net : what's wrong with the GPL? Just for my information. Thanks

**Comment by neilcsmith-net** [comment]
It's not Apache license compatible. For a full list of what can and cannot be included in an Apache project, license-wise, take a look at https://www.apache.org/legal/resolved.html

GPL is Category X, which means it cannot be included in source or binary formats.  GPL with CPE can be included in binary format given additional permissions (hence nb-javac is included), but straight GPL is a no go.

**Comment by albilu** [comment]
Full explanation: https://www.apache.org/licenses/GPL-compatibility.html
I am a FOSS advocate but that's the kind of things that piss me off :(

**Comment by neilcsmith-net** [comment]
The key thing is -

> The license, as applied in practice, must not impose significant restrictions beyond those imposed by the Apache License 2.0.

Being a FOSS advocate means respecting the terms under which people want to make their software available.  I release things personally under Apache License and under (L)GPL, and I expect people to respect the choice of terms for each project.  The author of WebLaf has chosen to add additional restrictions that exclude projects released under terms like NetBeans from using it.  That is their prerogative.

**Comment by albilu** [comment]
My comment was not against Netbeans at all! And i don't think the author of WebLaf chosed explicitly to restricted other projects.

As explained in the link above this is just a philosophy and interpretation issue. That's what piss me off.

> We avoid GPLv3 software because merely linking to it is considered by the GPLv3 authors to create a derivative work. We want to honor their license. Unless GPLv3 licensors relax this interpretation of their own license regarding linking, our licensing philosophies are fundamentally incompatible. This is an identical issue for both GPLv2 and GPLv3.

**Reply by neilcsmith-net** [reply]
Moving to discussion as this is becoming one! :smile: 

> And i don't think the author of WebLaf chosed explicitly to restricted other projects.

Yes, they did, and it's discussed at http://weblookandfeel.com/licensing/

It is absolutely not just a philosophy and interpretation issue.  It's about respecting the terms under which people make their code available.
